### PR TITLE
Split test app creation to a separate job

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
This is some initial implementation of https://github.com/activeadmin/activeadmin/pull/5530#issuecomment-431696455.

It turns out that the main reason why jruby is so slow is the creation of the test app, because at lot of subprocesses are spawned, and jruby has a very slow startup. This should alleviate that by creating a single test app per supported Rails version.

I haven't actually tried any of this in CI, will see how it goes once Github is back up.

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
This is now ready even if the WIP check won't go away for some reason!

File: .circleci/config.yml [hunk 1]
```diff
@@ -33,10 +33,25 @@ version: 2.1
     paths:
       - vendor/bundle
 
+.create_test_app: &create_test_app
```
Comment (javierjulio) [inline .circleci/config.yml]: I had to walk through this for a bit and in CI to double check but all makes sense. At first I had it wrong. Perhaps add a small note: "A test app is shared across Ruby versions but only for one Rails version."?

Comment (deivid-rodriguez) [reply .circleci/config.yml]: I added a general comment about the workflow in the workflows section in https://github.com/activeadmin/activeadmin/pull/5534/commits/61a17ce92eebdbb438ad8860d92cda6886609ec4, let me know what you think! Not very good at doc writing, but I did include some bullet points to try make it easier to read :smiley:.

Comment (javierjulio) [reply .circleci/config.yml]: That’s great thanks! 😄

**Comment by javierjulio** [comment]
@deivid-rodriguez I take it the new commit/push fixed the WIP issue too?

**Comment by deivid-rodriguez** [comment]
Apparently, yeah! Note that the WIP app has changed and now using the `work in progress` label is a [paid feature](https://github.com/wip/app/issues/89#issuecomment-432447327). So only changing the PR title will work to set a WIP status on the PR from now on.

**Comment by javierjulio** [comment]
Ok cool thanks!

**Pull request merged**